### PR TITLE
Define storage type as an env var

### LINKED ISSUE #61: Use SPAN_STORAGE env var instead of deprecated command line option
See https://github.com/jaegertracing/jaeger/pull/625#issuecomment-355851142

### PR DESCRIPTION
**Pull request by objectiser**
Fixes #61 

Signed-off-by: Gary Brown <gary@brownuk.com>

### CODE REVIEW

**Comment by pavolloffay** [comment]
I would rather keep prod template using a tagged version. 

Is all-in-one template failing due to changes in jaeger?

File: jaeger-production-template.yml [hunk 1]
```diff
@@ -48,6 +48,12 @@ items:
           volumeMounts:
           - name: jaeger-configuration-volume
             mountPath: /conf
+          env:
+          - name: SPAN_STORAGE
+            valueFrom:
+              configMapKeyRef:
+                name: jaeger-configuration
```
Comment (pavolloffay) [inline jaeger-production-template.yml]: How does it work it will set the value based on `configMap` key?

Comment (objectiser) [reply jaeger-production-template.yml]: Yes, it gets the value from the configmap, but sets it as an environment variable.

Comment (pavolloffay) [reply jaeger-production-template.yml]: nice feature :+1:

**Comment by objectiser** [comment]
@pavolloffay As mentioned in the PR comment - this should only be merged once jaeger 1.2 is released and the latest tag updated to 1.2. This is only set to latest currently to test the PR.

all-in-one is working locally for me - so will need to see if travis fails.

**Comment by objectiser** [comment]
@pavolloffay This should be fine to be merged now, after https://github.com/jaegertracing/jaeger/pull/643 fixed the backward compatibility issue by renaming the env var.

**Pull request merged**